src/cursor.c

```diff
@@ -14,26 +14,100 @@
 #include "common/scene-helpers.h"
 #include "common/zfree.h"
 
-static const struct cursor_alias {
-	const char *name, *alias;
-} cursor_aliases[] = {
-	{ "default",    "left_ptr" },
-	{ "text",       "xterm" },
-	{ "grab",       "grabbing" },
-	{ "pointer",    "hand1" },
-	{ "wait",       "watch" },
-	{ "all-scroll", "grabbing" },
-	/* Resize edges */
-	{ "nw-resize",  "top_left_corner" },
-	{ "n-resize",   "top_side" },
-	{ "ne-resize",  "top_right_corner" },
-	{ "e-resize",   "right_side" },
-	{ "se-resize",  "bottom_right_corner" },
-	{ "s-resize",   "bottom_side" },
-	{ "sw-resize",  "bottom_left_corner" },
-	{ "w-resize",   "left_side" },
+static const char **cursor_names = NULL;
+
+/* Usual cursor names */
+static const char *cursors_xdg[] = {
+	NULL,
+	"default",
+	"grab",
+	"nw-resize",
+	"n-resize",
+	"ne-resize",
+	"e-resize",
+	"se-resize",
+	"s-resize",
+	"sw-resize",
+	"w-resize"
 };
 
+/* XCursor fallbacks */
+static const char *cursors_x11[] = {
+	NULL,
+	"left_ptr",
+	"grabbing",
+	"top_left_corner",
+	"top_side",
+	"top_right_corner",
+	"right_side",
+	"bottom_right_corner",
+	"bottom_side",
+	"bottom_left_corner",
+	"left_side"
+};
+
+#define ARRAY_SIZE(array) (sizeof(array) / sizeof((array)[0]))
+static_assert(
+	ARRAY_SIZE(cursors_xdg) == LAB_CURSOR_COUNT,
+	"XDG cursor names are out ot sync");
+static_assert(
+	ARRAY_SIZE(cursors_x11) == LAB_CURSOR_COUNT,
+	"X11 cursor names are out ot sync");
+#undef ARRAY_SIZE
+
+static enum lab_cursors
+cursor_get_from_ssd(enum ssd_part_type view_area)
```

Comment:
In the interest of code-reuse, I suppose `cursor_get_from_edge(ssd_resize_edges(view_area))` would be equivalent, and save writing another function.

Reply:
Yes, thought about that but felt its unnecessary CPU work that has to be done which might be silly.
Would you be fine if I merge these two by having cursor_get_from_ssd call `ssd_resize_edges()` -> `cursor_get_from_edge()`?

Reply:
I'm fine with whatever you like best here, it was just an idea that popped into my head, and it sounds like you've already considered it :+1:

Reply:
Kept the wrapper but replaced the switch() with a call to `ssd_resize_edges()` and `cursor_get_from_edge()`